src/main/java/com/externalSorting/app/ExternalSortingRunner.java

```diff
@@ -9,60 +9,74 @@
-
-        System.out.println(outputFolder + "/Output.txt");
-        mergeFiles(files, new File("src/main/resources/SmallLibraryOutput.txt"), comparator);
+    /**
+     * This method is meant to be called from main function.
+     * It is responsible for:
+     *  creating a temporary folder which will hold data files
+     *  calling readAndSplitData and mergeFiles
+     *  deleting temporary folder and files after the process is finished
+     */
+    public void externalSort() {
+        File newDirectory = new File(outputFolder + "/tmpdir");
+        newDirectory.mkdir();
+        List<File> files = readAndSplitData(originalFile, newDirectory);
+        mergeFiles(files, new File(outputFolder + "/Output.txt"));
         for (File file : files) {
             file.delete();
         }
         newDirectory.delete();
     }
+
     /**
-    CHANGE_FILE
-    Maximum number of lines of text that can be put into a single temporary file
+    TMP_FILE_MAXSIZE
+    Maximum number of bytes of a single temporary file. The value of this variable controls memory usage.
     */
-    private static final int TMP_FILE_MAXSIZE = 24000000; // 24mb
+    private static final int TMP_FILE_MAXSIZE = 10000000; // around 10MB
 
     /**
     Defines a default string comparator
     */
-    private static final Comparator<String> comparator = (o1, o2) -> o1.compareTo(o2);
+    private static final Comparator<String> comparator = Comparator.naturalOrder();
 
     /**
-    Reads data from a single text file and splits it into a number of temporary text files
-    containing lists of alphabetically sorted words
-
-    *param* fileName
-    Path to original single text file to be read from
-
-    *param* directory
-    Path of the directory which will contain temporary text files with lists of sorted words
-
-    *return*
-    Returns a list of created temporary files containing sorted data from original file
-    */
-    private static List<File> readAndSplitData(String filePath, File directoryPath) {
+     * Reads data from a single text file and splits it into a number of temporary text files
+     * containing lists of alphabetically sorted words
+     *
+     * @param filePath
+     * Path to the text file from which data will be read
+     * @param directoryPath
+     * Path of the directory which will contain temporary text files with lists of sorted words
+     * @return
+     * Returns a list of created temporary files containing sorted data from original text file
+     */
+    private List<File> readAndSplitData(String filePath, File directoryPath) {
         List<String> buffer = new ArrayList<>();
         List<File> files = new ArrayList<>();
+        int NEWLINE_CHAR_SIZE = 1;
 
         try (BufferedReader reader = new MyBufferedReader(filePath)) {
             int currentEstimatedFileSize = 0;
+            List<String> splitLine;
             for (String line = reader.readLine(); line != null; line = reader.readLine()){
```

Comment:
I think there is a bug here. You have to debug and see if my understanding is correct but I think the problem here is that `for` loop calls `readLine()` twice (when starting the iteration and when finishing the iteration). Consider such set of data:

1 <-- this will be read in first iteration (first `readLine()` call)
1 <-- this will be read by second `readLine()` of first iteration and not used in the body of the loop
2 <-- this will be read in second iteration by first `readLine()`
3 <-- this will be read by second `readLine()` in second iteration and not used in the body of the loop

This way we omit every second item from the file.
When working with BufferedReader, it is better to work with `while` loops.

Reply:
<img width="1286" alt="Screen Shot 2021-06-14 at 3 41 04 PM" src="https://user-images.githubusercontent.com/84452361/121903074-39615c80-cd28-11eb-95e1-f17050855792.png">

In for loops the first statement is executed at the beginning of the block and only once.
Third statement, here its `line = reader.readLine()`, is called after every iteration. As you can see on the screenshot, the output is correct.